hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmMessageCallProcessor.java

```diff
@@ -78,6 +86,24 @@ public void start(final MessageFrame frame, final OperationTracer operationTrace
         }
     }
 
+    protected void executeHederaPrecompile(
+            final PrecompiledContract contract,
+            final MessageFrame frame,
+            final OperationTracer operationTracer,
+            final TokenAccessor tokenAccessor) {
+        if (contract instanceof EvmHTSPrecompiledContract htsPrecompile) {
```

Comment:
Since we pass the `MessageFrame` we can execute directly inside the original `executeHederaPrecompile` method like this:

`var updater = (HederaEvmStackedWorldStateUpdater) frame.getWorldUpdater();`

and thus get rid of the overloaded method.